arwen/contexts/output.go

```diff
@@ -328,8 +328,8 @@ func (context *outputContext) AddToActiveState(rightOutput *vmcommon.VMOutput) {
 		rightOutput.GasRefund.Add(rightOutput.GasRefund, context.outputState.GasRefund)
 	}
 
-	for _, rightAccount := range rightOutput.OutputAccounts {
-		leftAccount, ok := context.outputState.OutputAccounts[string(rightAccount.Address)]
+	for address, rightAccount := range rightOutput.OutputAccounts {
```

Comment:
please move it back :) it is much safer.

Reply:
Reverted.

Reply:
Using `string(rightAccount.Address)` will still cause problems if there are duplicate `OutputAccounts`, due to random iteration over maps. It doesn't actually help, it only hides the problem. The duplication was the issue, not this iteration.